### ISSUE TITLE
Class-Balanced Loss Based on Effective Number of Samples

### ISSUE THREAD

**Issue by ashutosh1919** [tensorflow/addons#2732]
Currently, TensorFlow has Focal loss which can be one of the options to handle the class imbalance issues. But I myself have faced problems in the case of some datasets like RecSys. IMO, TensorFlow should consist of Class-balanced loss ([paper](https://arxiv.org/pdf/1901.05555.pdf)).

I am interested in working on this issue. If you think we should add this, then please assign the issue to me and I can raise PR.

cc @bhack

**Comment by bhack** [comment]
As the focal loss [Is going to be duplicated in Keras-cv](https://github.com/keras-team/keras-cv/issues/319) can you open a ticket in keras-cv so that if they are interested we avoid another potential duplication?

**Comment by ashutosh1919** [comment]
Created [#16735](https://github.com/keras-team/keras/issues/16735)

**Comment by seanpmorgan** [comment]
TensorFlow Addons is transitioning to a minimal maintenance and release mode. New features will not be added to this repository. For more information, please see our public messaging on this decision:
[TensorFlow Addons Wind Down](https://github.com/tensorflow/addons/issues/2807)

Please consider sending feature requests / contributions to other repositories in the TF community with a similar charters to TFA:
[Keras](https://github.com/keras-team/keras)
[Keras-CV](https://github.com/keras-team/keras-cv)
[Keras-NLP](https://github.com/keras-team/keras-nlp)
